### PR TITLE
feat: nitro multi modal

### PR DESCRIPTION
**Pull request by tikikun**

### CODE REVIEW

File: controllers/llamaCPP.cc [hunk 1]
```diff
@@ -386,6 +437,11 @@ bool llamaCPP::loadModelImpl(const Json::Value &jsonBody) {
   int drogon_thread = drogon::app().getThreadNum() - 1;
   LOG_INFO << "Drogon thread is:" << drogon_thread;
   if (jsonBody) {
+    if (!jsonBody["mmproj"].isNull()) {
+      LOG_INFO << "MMPROJ FILE detected, multi-model enabled!";
+      params.mmproj = jsonBody["mmproj"].asString();
+      multi_modal = true;
```
Comment (louis-jan) [inline controllers/llamaCPP.cc:443]: toggle back on unload

Comment (tikikun) [reply controllers/llamaCPP.cc]: Resolve, now use llama internal state instead

**Comment by tikikun** [comment]
example of loading LLAVA:

```bash
curl -X POST 'http://localhost:3928/inferences/llamacpp/loadModel' \
-H 'Content-Type: application/json' \
-d '{
    "llama_model_path": "/Users/alandao/Downloads/ggml-model-q4_k.gguf","mmproj":"/Users/alandao/Downloads/mmproj-model-f16.gguf"
    ,"ctx_len": 4096,
    "ngl": 100, "cont_batching":false, "embedding":false}'
```

Example of doing inferencing with image

```python
import base64
import requests

# OpenAI API Key
api_key = "YOUR_OPENAI_API_KEY"

# Function to encode the image
def encode_image(image_path):
    with open(image_path, "rb") as image_file:
        return base64.b64encode(image_file.read()).decode('utf-8')


headers = {
    'Content-Type': 'application/json',
    'Authorization': f'Bearer {api_key}',
}

# Define the image_b64 variable here
image_b64 = encode_image('/Users/alandao/Downloads/download.jpeg')

json_data = {
    "model": "text-davinci-003",
    "messages": [
        {"role": "assistant", "content": "Hello there 👋"},
        {
            "role": "user",
            "content": [
                {"type": "text", "text": "Describe this image"},
                {
                    "type": "image_url",
                    "image_url": {
                        "url": f"data:image/jpeg;base64,{image_b64}"
                    },
                },
            ],
        },
    ],
    "stream": False,
}


def print_streaming_response(response):
    for chunk in response.iter_lines():
        if chunk:
            print(chunk.decode('utf-8'))


response = requests.post('http://localhost:3928/v1/chat/completions', headers=headers, json=json_data)
print_streaming_response(response)
```

**Comment by tikikun** [comment]
@vuonghoainam

**Comment by tikikun** [comment]
mode link
https://huggingface.co/mys/ggml_llava-v1.5-7b

**Pull request merged**